### PR TITLE
Lint and fix `bugprone-use-after-move` violations

### PR DESCRIPTION
**Pull request by dnfield**
Fixes https://github.com/flutter/flutter/issues/111135

Adds some ignores for tests that explicitly want to test behavior of use after move.

### CODE REVIEW

File: display_list/display_list_builder.cc [hunk 1]
```diff
@@ -1041,9 +1041,9 @@ void DisplayListBuilder::drawPicture(const sk_sp<SkPicture> picture,
                                      const SkMatrix* matrix,
                                      bool render_with_attributes) {
   matrix  //
-      ? Push<DrawSkPictureMatrixOp>(0, 1, std::move(picture), *matrix,
+      ? Push<DrawSkPictureMatrixOp>(0, 1, picture, *matrix,
                                     render_with_attributes)
-      : Push<DrawSkPictureOp>(0, 1, std::move(picture), render_with_attributes);
```
Comment (gaaclarke) [inline display_list/display_list_builder.cc:1046]: This seems like a false alarm.  Only one move is guaranteed to execute and this technically changes the performance for the worse.

File: impeller/aiks/canvas.cc [hunk 2]
```diff
@@ -336,7 +336,7 @@ void Canvas::SaveLayer(Paint paint,
 void Canvas::DrawTextFrame(TextFrame text_frame, Point position, Paint paint) {
   auto lazy_glyph_atlas = GetCurrentPass().GetLazyGlyphAtlas();
 
-  lazy_glyph_atlas->AddTextFrame(std::move(text_frame));
+  lazy_glyph_atlas->AddTextFrame(text_frame);
 
   auto text_contents = std::make_shared<TextContents>();
   text_contents->SetTextFrame(std::move(text_frame));
```
Comment (gaaclarke) [inline impeller/aiks/canvas.cc:342]: How is this not a bug?  This manifests as a bug right?

**Review by gaaclarke — COMMENTED** [review]
Awesome, LGTM.  I thought we had this on already.  I remember fixing the issue when I first implemented the linter.  My only concern was its inability to figure out the ternary operator case.  I don't want to accept worse performance to make the linter happy.

**Review by jason-simmons — APPROVED** [review]

File: display_list/display_list_builder.cc (hunk 1 above)
Comment (dnfield) [reply display_list/display_list_builder.cc:1046]: The problem isn't the ternary, it's that we reference picture on the line right under this.

File: impeller/aiks/canvas.cc (hunk 2 above)
Comment (dnfield) [reply impeller/aiks/canvas.cc:342]: @chinmaygarde may know. I don't understand how this ever worked :)

File: display_list/display_list_builder.cc (hunk 1 above)
Comment (gaaclarke) [reply display_list/display_list_builder.cc:1046]: Ahh missed that, nice catch. LGTM.

**Review by gaaclarke — APPROVED** [review]

**Comment by dnfield** [comment]
> Awesome, LGTM. I thought we had this on already. I remember fixing the issue when I first implemented the linter. My only concern was its inability to figure out the ternary operator case. I don't want to accept worse performance to make the linter happy.

There's some lint in there that is checking for calling methods on `std::move`d objects, but it doesn't appear to cover as many cases of use after move. `bugprone-use-after-move` is covering more. I thought of this because I was working on a patch where I was worried I might have introduced a use-after-move and started wondering if the linter was set up to catch those...

File: display_list/display_list_builder.cc (hunk 1 above)
Comment (dnfield) [reply display_list/display_list_builder.cc:1046]: (Well, the line under the comments under this, which GitHub is cutting off by default :)

**Comment by dnfield** [comment]
There have been a few violations introduced by new commits as I'm writing this PR. If it happens to fail in post submit we should fix forward rather than reverting.

The main culprit seems to be people copy/pasting something like `CreateShell(std::move(settings), std::move(task_runners))` from buggy tests that are getting fixed here.

**Comment by zanderso** [comment]
From PR triage: This is blocked on the tree closure @ https://github.com/flutter/flutter/issues/111193.

**Pull request merged**